### DISCUSSION TITLE
Pass a command line argument to the new terminal window that is about to be launched?

### DISCUSSION THREAD

**Discussion by RalfReddings** [microsoft/terminal#17521] [Q&A] [answered]
I am trying to figure out how to launch new terminal window and pass a command line to be executed by the new terminals shell (PowerShell 7)

The following works, a new terminal window, with Pwsh as its shell opens, and `get-childItem` is executed:
    
    wt pwsh -c get-childitem
    

But I want to choose what windows terminal profile to use for the new terminal window, rather than the generic Pwsh executable.

I tried using `-p` (`-profile`) and specifying my profile name:
    
    wt -p my_profile -c get-childitems
    
This causes windows terminal to show a dialog box with an error:
    
    The following argument was not expected: -c Run with --help for more information.
    

Running the following works just fine, the new window with the profile launches:
    
    wt -p my_profile
    

Is this even possible?

**Comment by DHowett** [comment, marked as answer]
Alas, this is one of the weird parts of Profiles as implemented in Terminal today.

The commandline passed on the command line (hah) is taken wholesale, and used to override the commandline of the profile you specify.

Right now, that means you need to do this:

```
wt -p my_profile my_shell -c get-childitem
```

There's an issue somewhere tracking changing (or clarifying, or extending?) this behavior, but I can't find it at the moment.

**Reply by RalfReddings** [reply]
hehe yh there is some inception involved. I can confirm that worked for me. Cheers!

**Reply by DHowett** [reply]
Glad it worked!
